src/providers/extraction_providers/visionExtractionProvider.ts

```diff
@@ -0,0 +1,77 @@
+import { ExtractionProvider } from './extractionProvider';
+import { Image, ExtractedImage } from '../../models/data_models/image';
+import { AssignedLabel, DetectedObject, Coordinate } from '../../models/data_models/extraction';
+
+import * as vision from '@google-cloud/vision';
+
+const enum requestTypes {
+	LabelDetection = 'LABEL_DETECTION',
+	ObjectLocalization = 'OBJECT_LOCALIZATION',
+}
+
+type LocalizedObjectAnnotation = vision.protos.google.cloud.vision.v1.ILocalizedObjectAnnotation;
+type EntityAnnotation = vision.protos.google.cloud.vision.v1.IEntityAnnotation;
+type NormalizedVertex = vision.protos.google.cloud.vision.v1.INormalizedVertex;
+
+export class VisionExtractionProvider implements ExtractionProvider {
+	protected createAssignedLabel(label: EntityAnnotation): AssignedLabel {
+		return {
+			description: label?.description || '',
+			confidence: label?.score || 0,
+		};
+	}
+
+	protected static createBoundingVertex(coordinate: NormalizedVertex): Coordinate {
+		return {
+			x: coordinate?.x || 0,
+			y: coordinate?.y || 0,
```

Comment:
This should probably use `??` instead of `||`, because the intention here is to use the right value if the left value is `undefined`. This doesn't matter right now (because the only `falsy` left value is `0`, so `0` (right value) would be used instead). However, if this situation ever changes (and to make it extra clear), `??` is clearer about intentions and future modifications.

For more reference on the difference between `||` and `??`, see https://stackoverflow.com/questions/61480993/when-should-i-use-nullish-coalescing-vs-logical-or